internal/kafka/internal/config/dataplane_cluster_config.go

```diff
@@ -92,8 +92,8 @@ const (
 
 // constants for Observability Operator installation via OpenShift Lifecycle Manager
 const (
-	defaultObservabilityOperatorIndexImage  = "quay.io/rhoas/observability-operator-index:v4.1.2"
-	defaultObservabilityOperatorStartingCSV = "observability-operator.v4.1.2"
+	defaultObservabilityOperatorIndexImage  = "quay.io/rhoas/observability-operator-index:v4.2.0"
```

Comment:
Is there any reason this is being updated in this PR? is it related?

Reply:
OO 4.2.0 is required for the proxy and since I needed to test this anyway, I've updated our default values too.

Reply:
Ah sorry, you're referring to the quay resources. Similar reason though, with OO 4.2.0 this now becomes the default way of fetching observability resources. So we should also udpate our defaults.